modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -741,19 +729,13 @@ def validate_form_526_location_codes(service_information)
         # only retrieve separation locations if we'll need them
         invalid_end_date = detect_invalid_active_duty_enddate(service_information).is_a?(Array)
 
-        need_locations = service_information['servicePeriods'].detect do |service_period|
-          if invalid_end_date && service_period['activeDutyEndDate']
-            Date.strptime(service_period['activeDutyEndDate'],
-                          '%Y-%m-%d') > Time.zone.today
-          end
+        need_locations = service_information['servicePeriods'].any? do |service_period|
+          service_period['separationLocationCode'].present?
         end
         separation_locations = retrieve_separation_locations if need_locations
 
         service_information['servicePeriods'].each do |service_period|
-          if invalid_end_date && (service_period['activeDutyEndDate'] &&
-            Date.strptime(service_period['activeDutyEndDate'], '%Y-%m-%d') <= Time.zone.today)
-            next
-          end
+          next if invalid_end_date
```

Comment:
As discussed, I'm not sure we're actually validating anything here since this method neither raises an exception nor calls `collect_error_messages`. To be investigated in a future task.